File: truth_set/python/structure/solution-approach/semantic-duplication/services/userService.py
from datetime import datetime
from ..utils.dateUtils import format_date

class UserService:
    def __init__(self):
        self.users = []
    
    def add_user(self, name):
        new_user = {
            'id': len(self.users) + 1,
            'name': name,
            'registration_date': datetime.now()
        }
        self.users.append(new_user)
        print(f"User Added: {new_user['name']}, Registered On: {format_date(new_user['registration_date'], 'MM/DD/YYYY')}")
    
    def list_users(self):
        for user in self.users:
            print(f"ID: {user['id']}, Name: {user['name']}, Registration Date: {format_date(user['registration_date'])}")
\ No newline at end of file

File: truth_set/go/structure/solution-approach/semantic-duplication/main/services/userService.go
package services

import (
	"fmt"
	"time"
	"main/utils"
)

type User struct {
	ID               int
	Name             string
	RegistrationDate time.Time
}

type UserService struct {
	users []User
}

func NewUserService() *UserService {
	return &UserService{users: make([]User, 0)}
}

func (s *UserService) AddUser(name string) {
	newUser := User{
		ID:               len(s.users) + 1,
		Name:             name,
		RegistrationDate: time.Now(),
	}
	s.users = append(s.users, newUser)
	fmt.Printf("User Added: %s, Registered On: %s\n",
		newUser.Name, utils.FormatDate(newUser.RegistrationDate, "MM/DD/YYYY"))
}

func (s *UserService) ListUsers() {
	for _, user := range s.users {
		fmt.Printf("ID: %d, Name: %s, Registration Date: %s\n",
			user.ID, user.Name, utils.FormatDate(user.RegistrationDate, "YYYY-MM-DD"))
	}
}
\ No newline at end of file